mgnify_pipelines_toolkit/analysis/shared/study_summary_generator.py

```diff
@@ -0,0 +1,263 @@
+                    index_label="taxonomy",
+                )
+
+
+def summarise_analyses(runs_df, analyses_dir, output_prefix):
+
+    all_db_labels = TAXDB_LABELS + ASV_TAXDB_LABELS
+    for db_label in all_db_labels:
+
+        tax_files = defaultdict(Path)
+        for i in range(0, len(runs_df)):
+            run_acc = runs_df.loc[i, "run"]
+            tax_file = get_tax_file(run_acc, analyses_dir, db_label)
+
+            if tax_file:
+                tax_files[run_acc] = tax_file
+
+        if (
+            len(tax_files) > 1
+        ):  # If at least two analyses have results from the current DB, generate a study-level summary for it
+            generate_db_summary(db_label, tax_files, output_prefix)
+
+
+def organise_study_summaries(all_study_summaries):
+
+    summaries_dict = defaultdict(list)
+
+    for summary in all_study_summaries:
+        summary_path = Path(summary)
+        summary_filename = summary_path.stem
+
+        temp_lst = summary_filename.split("_")
+        if "asv_study_summary" in summary_filename:
+            summary_db_label = "_".join(
+                temp_lst[1:3]
+            )  # For ASVs we need to include the amp_region in the label
+        else:
+            summary_db_label = temp_lst[
+                1
+            ]  # For closed reference, just the db_label is needed
+
+        summaries_dict[summary_db_label].append(summary_path)
+
+    return summaries_dict
+
+
+def merge_summaries(analyses_dir, output_prefix):
+
+    # TODO: The way we grab all the summaries might change depending on how the prefect side does things
+    all_study_summaries = glob.glob(f"{analyses_dir}/*_study_summary.tsv")
+
+    summaries_dict = organise_study_summaries(all_study_summaries)
+
+    for db_label, summaries in summaries_dict.items():
+        if len(summaries) > 1:
+            res_df = pd.read_csv(summaries[0], sep="\t", index_col=0)
+            for summary in summaries[1:]:
+                curr_df = pd.read_csv(summary, sep="\t", index_col=0)
+                res_df = res_df.join(curr_df, how="outer")
+                res_df = res_df.fillna(0)
+                res_df = res_df.astype(int)
+
+            res_df = res_df.reindex(sorted(res_df.columns), axis=1)
+            res_df.to_csv(
+                f"{output_prefix}_{db_label}_study_summary.tsv",
+                sep="\t",
+                index_label="taxonomy",
+            )
+
+
+def main():
+
+    indir, runs, mode, prefix = parse_args()
+
+    if mode == "summarise":
+        if not runs:
+            logging.error(
+                "Can't run in `summarise` mode without specifying -r/--runs - exiting."
+            )
+            exit(1)
+
+        runs_df = pd.read_csv(runs, names=["run", "status"])
+        summarise_analyses(runs_df, indir, prefix)
+    elif mode == "merge":
+        merge_summaries(indir, prefix)
+    else:
+        logging.error("Mode can only be `summarise` or `merge` - exiting.")
+        exit(1)
```

Comment:
I don't think this case can ever happen given the argparse choices would catch this.
```suggestion
```